localstack/services/s3/utils.py

```diff
@@ -200,6 +200,16 @@ def get_bucket_and_key_from_s3_uri(s3_uri: str) -> Tuple[str, Optional[str]]:
     return output_bucket, output_key
 
 
+def get_bucket_and_key_from_presign_url(presign_url: str) -> Tuple[str, Optional[str]]:
+    """
+    Extracts the bucket name and key from s3 presign url
+    """
+    parsed_url = urlparser.urlparse(presign_url)
+    bucket = parsed_url.path.split("/")[1]
+    key = "/".join(parsed_url.path.split("/")[2:]).split("?")[0]
+    return bucket, key
```

Comment:
Under what conditions would `key` be `None` (as indicated in the return hint)

Reply:
@viren-nadkarni eg. the presign url is: http://s3.localhost.localstack.cloud:4566/test-output-bucket-2?AWSAccessKeyId=000000000000&Signature=2Yc%2BvwhXx8UzmH8imzySfLOW6OI%3D&Expires=1688561914

Reply:
It was already added in the [latest commit](https://github.com/localstack/localstack/pull/8619/commits/2fb35cb63fa7cb7d672604f856bbb09cc872020c). Is something else also required? 🤔